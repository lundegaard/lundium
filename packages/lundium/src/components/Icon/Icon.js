import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import { stringOrNumberPropType } from '../../utils/types';
import Text from '../Text';

const Icon = forwardRef(
	({ children, type, size = '14', className, color = 'dark', ...rest }, ref) => (
		<Text
			as="i"
			{...rest}
			color={color}
			className={cx(
				'icon',
				`icon--type-${type}`,
				`icon--size-${size}`,
				`color-${color}`,
				className
			)}
			ref={ref}
		>
			{children}
		</Text>
	)
);

Icon.propTypes = {
	...Text.propTypes,
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	/** Color of icon. */
	// FIXME - export from scss is empty for jest tests
	// color: PropTypes.oneOf(colors),
	color: PropTypes.string,
	/** Size of icon. */
	size: stringOrNumberPropType,
	/** Type of icon. */
	type: PropTypes.string.isRequired,
};

Icon.displayName = 'Icon';

export default Icon;
