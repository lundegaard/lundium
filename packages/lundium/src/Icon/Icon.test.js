import React from 'react';
import { shallow } from 'enzyme';

import Icon from './Icon';

// jest.mock('./Icon.scss', () => ({
// 	icons: 'add',
// }));

describe('<Icon />', () => {
	const requiredProps = { type: 'add' };

	it('shallow', () => {
		const wrapper = shallow(<Icon {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
