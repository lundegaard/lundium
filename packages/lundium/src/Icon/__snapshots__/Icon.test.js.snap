// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> shallow 1`] = `
<forwardRef(Text)
  as="i"
  className="icon icon--type-add icon--size-14 color-dark"
  color="dark"
/>
`;
